### PR TITLE
Reset axes extent and limits post transpose orthog data

### LINKED ISSUE #30446: Sliceviewer not plotting data correctly when axes transposed
### Actual behavior
Depending on the level of zoom and whether the peak is centred in the plot, when the axes are swapped the peak is shifted.

### Steps to reproduce the behavior
(1) run this script to generate fake data and a peak at (1,1,1)
```
ws = CreateMDWorkspace(Dimensions='3', Extents='-5,5,-5,5,-5,5',
                       Names='Q_lab_x,Q_lab_y,Q_lab_z', Units='U,U,U',
                       Frames='QLab,QLab,QLab',
                       SplitInto='2', SplitThreshold='50')
expt_info = CreateSampleWorkspace()
ws.addExperimentInfo(expt_info)
FakeMDEventData(ws, EllipsoidParams='1e+06,1,1,1,1,0,0,0,1,0,0,0,1,0.01,0.04,0.02,1', RandomSeed='3873875')
# add peak
p = CreatePeaksWorkspace(InstrumentWorkspace='ws', NumberOfPeaks=0, OutputWorkspace='peaks')
SetUB('peaks', 1,1,1,90,90,90)
pk = p.createPeak([1,1,1])
p.addPeak(pk)
```
(2) Open sliceviewer and overlay the peaks workspace
(3) Zoom in on the peak and swap the X and Y axes - the peak might shift (it seems to be more obvious if the peak is not centred in the plot)

image

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**

When transposing orthogonal data, the sliceviewer needs to update the axes extent and limits to acount for the transposition - this was responsible for the peak apparently moving when the axes were transposed. This is not a problem for non-orthogonal data where the axes limits are reset anyway.

**To test:**
See instructions in original issue.
- The peak at (1,1,1) should remain at (X,Y) = 1 not matter which axes are being transposed.

Fixes #30446

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@RichardWaiteSTFC Looks like some conflicts have appeared her

**Comment by joseph-torsney** [comment]
Pretty sure the changes work but I'm unable to reproduce the original issue on master due to build issues, so un-assigning myself for now while I fix them. Feel free anyone else to pick this review up in the meantime.

File: qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_view.py [hunk 1]
```diff
@@ -30,21 +30,22 @@
...
+                                              Units='MomentumTransfer,EnergyTransfer,Angstrom',
+                                              OutputWorkspace='ws_MD_2d')
+        cls.hkl_ws = CreateMDWorkspace(Dimensions=3,
+                                       Extents='-10,10,-9,9,-8,8',
+                                       Names='A,B,C',
+                                       Units='r.l.u.,r.l.u.,r.l.u.',
+                                       Frames='HKL,HKL,HKL',
+                                       OutputWorkspace='hkl_ws')
         expt_info = CreateSampleWorkspace()
         self.hkl_ws.addExperimentInfo(expt_info)
```
Comment (joseph-torsney) [inline qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_view.py:50]:
```suggestion
        cls.hkl_ws.addExperimentInfo(expt_info)
```
This should fix the flake8 warning

**Review by joseph-torsney — CHANGES_REQUESTED** [review]
Everything looks good but flake8!

File: qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_view.py (hunk 1 above)
Comment (RichardWaiteSTFC) [reply qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_view.py:50]: Thanks, that got introduced when I fixed the merge conflict on github, I was going to fix it but was on another branch so thanks for saving me the time!

**Review by joseph-torsney — APPROVED** [review]
Nice work :+1:

**Review by martyngigg — APPROVED** [review]
I just checked a MatrixWorkspace too and that looks good also 👍🏻

**Pull request merged**